db/migrate/20181001192951_make_user_names_case_insensitive.rb

```diff
@@ -0,0 +1,12 @@
+class MakeUserNamesCaseInsensitive < ActiveRecord::Migration[5.2]
+  def up
+    enable_extension :citext
+    change_column :users, :name, :citext
+    remove_index :users, :name
+    add_index :users, :name, unique: true
+  end
+
+  def down
+    change_column :users, :name, :text
```

Comment:
Is there a reason to not put the state of the db back exactly the way it was before?